sdk/dotnet/Pulumi/Deployment/Deployment_Prepare.cs

```diff
@@ -52,6 +52,28 @@ public partial class Deployment
                 providerRef = await ProviderResource.RegisterAsync(customOpts?.Provider).ConfigureAwait(false);
             }
 
+            Dictionary<string, string>? providerRefs = null;
+            if (remote)
+            {
+                var componentOpts = options as ComponentResourceOptions;
+                if (componentOpts != null)
+                {
+                    // If only the Provider opt is set, move it to the Providers list for further processing.
+                    if (componentOpts.Provider != null && componentOpts.Providers.Count == 0)
+                    {
+                        componentOpts.Providers.Add(componentOpts.Provider);
+                        componentOpts.Provider = null;
+                    }
+
+                    providerRefs = new Dictionary<string, string>(componentOpts.Providers.Count);
+                    foreach (var provider in componentOpts.Providers)
+                    {
+                        var pref = await ProviderResource.RegisterAsync(provider).ConfigureAwait(false);
+                        providerRefs.Add(provider.Package, pref ?? "");
```

Comment:
If `pref` is `null`, should we avoid adding it to the dictionary?

```suggestion
if (pref != null)
{
    providerRefs.Add(provider.Package, pref);
}
```

(Though, looking at `ProviderResource.RegisterAsync`, the result is only `null` if the passed-in `provider` is `null`, so the null case is unlikely to happen unless someone was able to get a `null` instance in the `componentOpts.Providers` list, but if so, we'd already hit a null ref when `provider.Package` is referenced).